torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1735,10 +1738,32 @@ def backward(ctx, *flat_args):
                 # TODO: figure out how to refactor the backward properly
                 # so I can use aot_dispatch_subclass_wrapper() here.
                 if CompiledFunction.maybe_subclass_metadata is not None:
+                    tangents = all_args[tangents_start_idx:tangents_end_idx]
+                    runtime_subclass_tangent_meta = create_subclass_meta(
```

Comment:
oh I see. I would prefer to not create fresh `SubclassCreationMeta`s at runtime here:

(1) this is all "hot-path" code, and any python data structures we create at runtime are just extra overhead

(2) AOTAutograd already has a lot of metadata, and knowing that all metadata is statically created at runtime is a nice invariant.

Reply:
alternative: the `SubclassCreationMeta` stashes the type of the current subclass as well as its constant metadata, so you can just compare that directly against the runtime subclasses that enter the backward.